src/service-worker.js

```diff
@@ -31,52 +31,53 @@ self.addEventListener('activate', event => {
 	);
 });
 
+
+async function fetchAndCache(request) {
+	const cache = await caches.open(`offline${timestamp}`)
+
+	try {
+		const response = await fetch(request);
+		cache.put(request, response.clone());
+		return response;
+	} catch (err) {
+		const response = await cache.match(request);
+		if (response) return response;
```

Comment:
we only return from the cache if there's an exception while fetching? Is that some peculiarity of one of the service worker APIs or something?

I'd have expected the code to look more like:
```
	let response = await cache.match(request);
	if (response) return response;
	response = await fetch(request);
	cache.put(request, response.clone());
	return response;
```

Reply:
These aren't static assets, so I guess they could change at any moment and this code only provides offline support. There's a comment saying
```
				// for everything else, try the network first, falling back to
				// cache if the user is offline. (If the pages never change, you
				// might prefer a cache-first approach to a network-first one.)
```

But I haven't changed any of the logic apart from the issue of not returning a cached value if it's not in fact present.

Reply:
ah, got it. thanks

Reply:
It might be good to move that comment here where the method is defined:

```
/**
 * Try the network first, falling back to cache if the user is offline. (If the pages never change,
 * you might prefer a cache-first approach to a network-first one.)
 */
```

I also wonder if we should rename the function to something like `fetchDynamicContent` that makes it clearer that it's doing something different than the static case

Reply:
It's true it's still a bit confusing. I restructured it a bit more, which I think makes it clearer, pulling out a `serveFromCache` variable that determines what is "cache first" and what isn't. 

But `fetchAndCache` will be used even for `serveFromCache` content (which is static) so I don't think referencing "dynamic" is right.

(There are essentially three levels of caching you can choose from here: preloaded, served from cache or "fetched and fall back on cache for offline")